src/main/java/net/datafaker/idnumbers/EsMXIdNumber.java

```diff
@@ -47,26 +49,24 @@ public class EsMXIdNumber implements IdNumbers {
      */
     public String get(BaseProviders faker) {
 
-        String sex = new String[]{"H", "M"}[faker.random().nextInt(2)];
+        final Options options = faker.options();
         String birthDay = getBirthday(faker);
-        String state = faker.options().option(STATES);
 
-        String c1 = faker.options().option(CONSONANT);
-        String c2 = faker.options().option(CONSONANT);
-        String c3 = faker.options().option(CONSONANT);
 
-        String v1 = faker.options().option(VOWEL);
-        String v2 = faker.options().option(VOWEL);
-        String v3 = faker.options().option(VOWEL);
-        String v4 = faker.options().option(VOWEL);
-
-        String ranNum = (Integer.parseInt(birthDay.substring(0, 4)) <= 1999) ?
-            "0" : faker.options().option(CONSONANT);
-
-        String ssn = c1 + v1 + c2 + c3 + birthDay.substring(2, 8) + sex + state + v2 + v3 + v4 + ranNum;
-        ssn = ssn + getChecksum(ssn);
-
-        return ssn;
+        String ssn = options.option(CONSONANT) +
```

Comment:
Would this chunk be any faster with a StringBuilder using append?

Reply:
It seems there is even faster way.
Since the length of the result is always same there could be used just array of chars without any concatenation